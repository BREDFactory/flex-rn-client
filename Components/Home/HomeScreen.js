// @flow

import React from 'react';
import { View, Button, AsyncStorage } from 'react-native';
import { NavigationScreenProp } from "react-navigation";
import styles from "./HomeScreenStyles";

type Props = {
  navigation: NavigationScreenProp<{}>
};

class HomeScreen extends React.Component<Props> {
  static navigationOptions = {
    title: 'Home'
  };

  componentDidMount() {
    const { navigate } = this.props.navigation;

    AsyncStorage.getItem('USER', (err, result) => {
      if (err || result == null) navigate('Login');
      else {
        const jsonres = JSON.parse(result)
        if (jsonres.place === null || jsonres.place === '') navigate('Profile');
        else navigate('Leave');
      }
    })
  }

  render() {
    return (
      <View style={styles.view}>
        <Button
          style={styles.button}
          title="Main Page"
          color="#000"
          onPress={() => this.componentDidMount()}
        />
      </View>
    )
  }
}

export default HomeScreen
