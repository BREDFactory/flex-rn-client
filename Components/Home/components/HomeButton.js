import { Button, View } from 'react-native'
import styles from '../HomeScreenStyles'
import I18n from '../../../i18n/i18n'
import React from 'react'

const HomeButton = (props: { onPress: () => void }) => {
  return (
    <View style={styles.button_wrapper}>
      <Button
        style={styles.button}
        title={I18n.t('home.main_page')}
        color="#5167A4"
        onPress={props.onPress}
      />
    </View>)
}

export default HomeButton
