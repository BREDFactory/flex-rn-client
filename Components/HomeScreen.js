import React from 'react';
import { TouchableOpacity, View, Text, Button } from 'react-native';
import { AsyncStorage } from 'react-native';

class HomeScreen extends React.Component {
  static navigationOptions = {
    title: 'Home',
  };

  componentDidMount(){
    const navigate = this.props.navigation.navigate;

    AsyncStorage.getItem('USER', (err, result) => {
      if(err || result == null)
        navigate('Login');
      else
      {
        var jsonres = JSON.parse(result);
        if(jsonres.place == null || jsonres.place == '')
          navigate('Profile');
        else
          navigate('Leave');
      }
    });
  }

  render() {
    const navigate = this.props.navigation.navigate;

    return (
      <View style={{
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
      }}>

        <Button
          title='Main Page'
          color="#000"
          onPress={() => this.componentDidMount()}/>
      </View>
    )
  }
}

export default HomeScreen;
