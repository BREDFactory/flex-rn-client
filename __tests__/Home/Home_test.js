import * as enzyme from 'enzyme';
import 'react-native';
import React from 'react';
import HomeScreen from '../../Components/Home/HomeScreen';
import HomeButton from '../../Components/Home/components/HomeButton';

import renderer from 'react-test-renderer';

import ReactSixteenAdapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new ReactSixteenAdapter() });

const navigation = { navigate: jest.fn(), popToTop: jest.fn() };

describe('Testing HomeScreen component', () => {
  it('renders as expected', () => {

    const wrapper = enzyme.shallow(
      <HomeScreen navigation={navigation} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});