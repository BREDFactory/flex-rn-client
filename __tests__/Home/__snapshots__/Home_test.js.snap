// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderColor": "#5167A4",
        "borderRadius": 5,
        "borderWidth": 2,
      }
    }
  >
    <View
      accessibilityComponentType="button"
      accessibilityTraits={
        Array [
          "button",
        ]
      }
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#5167A4",
              },
            ]
          }
        >
          Page Principale
        </Text>
      </View>
    </View>
  </View>
</View>
`;
